### ISSUE TITLE
DMX Skull Model doesn't work as expected

### ISSUE THREAD

**Issue by kbostick88** [smeighan/xLights#3159]
**Describe the bug**
When I add the servo effect on the DMXSkull model and set the base channel to Jaw/Eyes Up/Eyes Left/Nod/Tilt/Pan all do not move the skull model as expected. If you change the Base channel to one of these options and adjust the value, you can see in the preview the head is not responding correctly.

When Assigned to a controller and output to lights is on, it doesn't control the servos at all. If I use a block channel and apply an on effect, adjusting the brightness controls the servos as expected. 

**To Reproduce**
See attached video

**Expected behavior**
I expect the option selected to control the option described. Jaw - Jaw, Nod - Nod etc. 

**Attachments**

https://user-images.githubusercontent.com/17733692/158237308-a754c3b5-28c3-4d38-854a-3666758d79c7.mp4

[test skull.zip](https://github.com/smeighan/xLights/files/8247584/test.skull.zip)

**Comment by AzGilrock** [comment]
I'm at work so just a quick look at your rgbeffects file and the video my guess is this is just a mismatch between the channel assignments you've given and the Node Names fields.  Also I believe the model shows it set as a Skulltronix skull but you've changed the channel assignments from what I believe that skull uses.  The servo names that show up in that Servo Effect dropdown are in your control.  You create them with the Node Names field for the model.  So if you change the channel assignments then you  needed to also fix the node names to match or it won't be controlling the channel you think it does.  For a Skulltronix model I filled in all the defaults so it would work but once you start changing channel assignments your also responsible for fixing the node names.  You'll notice I use a minus sign in front of a name that I don't want to show up so that help when you have a 16 bit servo I have the channels listed as Jaw and -Jaw and only Jaw will show up.

**Comment by AzGilrock** [comment]
Anyways I need to double check things at home.  I thought I remembered fixing it so that the node names automatically updated but maybe I never got that feature done.  It may work better if you uncheck the Skulltronix checkbox in the dialog where you define which servos are used and then double-check the node names.

**Comment by kbostick88** [comment]
Thanks Gil, I will check that out and see if I can change the assignments if it makes a difference. i'll keep you posted.

**Comment by kbostick88** [comment]
I updated the names but it is still doing the same thing. I also tried to turn off the skulltronix skull, same results. This could be a configuration issue on my end, or a bug I am not 100% sure. The people I've asked about this aren't very familiar with this model or its set up so they were not able to help.

**Comment by AzGilrock** [comment]
Well the easy test is drop a brand new one down and see how it works without touching any channels.

**Comment by kbostick88** [comment]
If you need/want me to jump into a zoom i'd be happy to. just let me know :)

**Comment by AzGilrock** [comment]
It was working back when I wrote it so I'm basically just testing whether the other guys broke anything. ;)

**Comment by dkulp** [comment]
The I Like Nightmares thing you sent me seems to be working fine.   :)

**Comment by AzGilrock** [comment]
Yeah I just got home and tested the default skull and its working fine.  I did test swapping the Jaw and Pan channel definitions and it made the servo effect work backwards but that's because the Node names are setup manually so that can be adjusted by the user to fix it.

**Comment by kbostick88** [comment]
I added a brand new dmx skull and tried that, it seems to work better until I adjust some of the servo values. Once I adjust the some of the servos the jaw basically appears above its head. 

In order to verify this isn’t an issue on my end (something I’m doing wrong) is there a guide/manual/video on how to set this up properly? I looked in the xlights manual but it isn’t in there.

**Comment by AzGilrock** [comment]
The other thing with the Skull model is you really need to create a model layer for each servo because if they aren't all defined the model can react weird.  It needs to have input for all the channels at least for the movement.

**Comment by AzGilrock** [comment]
I don't have my own Zoom...maybe if we jumped on the xLights one they could make a room for us

**Comment by kbostick88** [comment]
I can jump in the main room, or I have an account and could share a room code. Whatever you guys prefer

**Comment by AzGilrock** [comment]
I'm in there

**Comment by AzGilrock** [comment]
We worked on this in Zoom and got everything working.  I did notice Kyle needed a Jaw Orientation parameter because his Jaw motion worked backwards and also the skull was disappearing when he reduced his channel count because he wasn't using the color channels.  So I just checked in an update that adds the Jaw Orientation parameter and I fixed the color channel error checking to only occur when color channels are active.
